exercise_2/src/exercise_2.cpp

```diff
@@ -0,0 +1,61 @@
+#include <stdio.h>
+#include <limits.h>
+#include <iostream>
+#include <fstream>
+
+using namespace std;
+
+int const VERTICES {8};
+int const ROOT {0};
+
+int minDistance(int distance[], bool isVisited[])
+{
+   int min = INT_MAX, min_index;
+   for (int v = 0; v < VERTICES; v++)
+     if (isVisited[v] == false && distance[v] <= min)
+         min = distance[v], min_index = v;
+   return min_index;
+}
+
+void findShortestPath(int graph[VERTICES][VERTICES], int src)
```

Comment:
A couple of mistakes here:
- complex types should not be copied as it is in this case. I mean - an array is not a complex type, but in this case it will be copied.
- don't define size of arrays in function. In theory we could reuse this function somewhere else. What you should do is to pass a const reference to something that stores the graph data - a vector of node connection representation?
If you really want to use two dimensional arrays then you should define this function as:
void findShortestPath(int graph[][], int columns, int rows, int src)
- this fuction is called findShortestPath but it returns void. If a job of a function is to find something then it should somehow return the result using either return statement or output parameters (arguments passed by a reference)

Reply:
Changed.